packages/react-reconciler/src/ReactInternalTypes.js

```diff
@@ -247,7 +247,10 @@ type BaseFiberRootProperties = {|
   // a reference to.
   identifierPrefix: string,
 
-  onRecoverableError: (error: mixed) => void,
+  onRecoverableError: (
+    error: mixed,
+    errorInfo: {errorHash?: ?string, componentStack?: ?string},
```

Comment:
This is the first place we actually name `errorHash` anything in a public API. It's intended to be a hash but you could use it in other ways too (including just returning the full error message in production). I wonder if we should call it something more generic?

Reply:
`errorCode`?

It could be a generated ID too. Doesn't need to be a hash.

Reply:
I like `errorCode`, feels less prescriptive than hash or id. similarly `errorDigest` evokes some potentially summarized error data and has analogs with crypto hashing as the output so it works kinda however you use it. 

Code perhaps implies codification too much

Reply:
One downside of `errorCode` is that it's often associated with [the minified error codes](https://github.com/facebook/react/blob/main/scripts/error-codes/codes.json).

Maybe that should actually be a special field too for minified errors so you don't have to parse it yourself. Even for client errors.

(If you don't provide onError on the server and it's a React error, maybe we should use the minified error code as the default "hash" too.)

Reply:
So

```
{
  // For React errors in Prod Server or Client: the React Error Code
  code?: string,
  
  // For Server user errors: onError return value
  // For Server React errors if no onError provided: the React Error Code
  digest? string, // or whatever this name becomes
  
  // For Dev Server errors (React or user): the server component stack
  // For Dev or Prod Client errors (React or user): the client component stack
  componentStack?: string,
}
```

Reply:
Sounds good. Let's do the `code` part later though.